src/test/java/com/teragrep/pth_06/planner/CategoryTableImplTest.java

```diff
@@ -0,0 +1,312 @@
+                    + "    `filter`         longblob            NOT NULL)";
+            conn.prepareStatement(targetTable).execute();
+        });
+    }
+
+    @AfterAll
+    void tearDown() {
+        Assertions.assertDoesNotThrow(() -> {
+            conn.prepareStatement("DROP ALL OBJECTS").execute(); //h2 clear database
+            conn.close();
+        });
+    }
+
+    @Test
+    public void testNonCreatedEmptyTable() {
+        DSLContext ctx = DSL.using(conn);
+        Table<?> table = ctx
+                .meta()
+                .filterSchemas(s -> s.getName().equals("bloomdb"))
+                .filterTables(t -> !t.getName().equals("filtertype"))
+                .getTables()
+                .get(0);
+
+        TokenizedValue val = new TokenizedValue("test");
+        Assertions.assertDoesNotThrow(new CategoryTableImpl(ctx, table, 0L, val)::bloommatchCondition);
+    }
+
+    @Test
+    public void testCreatedWithEmptyTable() {
+        DSLContext ctx = DSL.using(conn);
+        Table<?> table = ctx
+                .meta()
+                .filterSchemas(s -> s.getName().equals("bloomdb"))
+                .filterTables(t -> !t.getName().equals("filtertype"))
+                .getTables()
+                .get(0);
+
+        TokenizedValue val = new TokenizedValue("test");
+        CategoryTable tempTable = new Created(new WithFilterTypes(new CategoryTableImpl(ctx, table, 0L, val)));
+        RuntimeException ex = Assertions.assertThrows(RuntimeException.class, tempTable::bloommatchCondition);
+        Assertions.assertEquals("Origin table was empty", ex.getMessage());
+    }
+
+    @Test
+    public void testCreation() {
+        fillTargetTable();
+        DSLContext ctx = DSL.using(conn);
+        Table<?> table = ctx
+                .meta()
+                .filterSchemas(s -> s.getName().equals("bloomdb"))
+                .filterTables(t -> !t.getName().equals("filtertype"))
+                .getTables()
+                .get(0);
+
+        TokenizedValue val = new TokenizedValue("test");
+        CategoryTable created = new Created(new CategoryTableImpl(ctx, table, 0L, val));
+        Assertions
+                .assertEquals(created.bloommatchCondition(), new CategoryTableImpl(ctx, table, 0L, val).bloommatchCondition());
+    }
+
+    @Test
+    public void testConditionGeneration() {
+        fillTargetTable();
+        DSLContext ctx = DSL.using(conn);
+        Table<?> table = ctx
+                .meta()
+                .filterSchemas(s -> s.getName().equals("bloomdb"))
+                .filterTables(t -> !t.getName().equals("filtertype"))
+                .getTables()
+                .get(0);
+
+        TokenizedValue val = new TokenizedValue("test");
+        CategoryTableImpl tempTable = new CategoryTableImpl(ctx, table, 0L, val);
+        QueryCondition tableCond = tempTable.bloommatchCondition();
+        QueryCondition expectedCond = new CategoryTableCondition(table, 0L);
```

Comment:
Would it be better to create expected queryCondition "manually" instead of relying on another class?
If both classes make the same wrong result the test would pass

Reply:
Refactored to compare queryCondition using condition.toString() like in other unit tests